internal/devpkg/package.go

```diff
@@ -284,8 +284,25 @@ func (p *Package) NormalizedPackageAttributePath() (string, error) {
 	return p.normalizedPackageAttributePathCache, nil
 }
 
+// search attempts to find the package in the local nixpkgs. It may be an expensive
+// call, if it is has not been cached.
+func (p *Package) search(query string) (map[string]*nix.Info, error) {
+	if p.IsDevboxPackage() {
+		if strings.HasPrefix(query, "runx:") {
+			// TODO implement runx search
+			return map[string]*nix.Info{}, nil
+		}
+		// This will be slow if its the first time on the user's machine that this
+		// query is running. Otherwise, it will be cached and fast.
+		return nix.SearchNixpkgsAttribute(query)
+	}
+
+	// fallback to the slow but generalized nix.Search
+	return nix.Search(query)
```

Comment:
Why the distinction between devbox packages and flakes? Can't you cache the flakes as well?

Reply:
We cannot use `nix.Search` caching for flakes, local flakes at least. For now, I'd lean towards being conservative and just support queries of the form `nixpkgs/commit#attribute`

To illustrate the problem, consider the `examples/flakes/php`. This has a package ` "path:my-php-flake#php"`

The `query` for this package is of the form: `<absolute path on laptop>/devbox/examples/flakes/php/my-php-flake#php`

Originally, that flake exported:
```
php = pkgs.php.withExtensions ({ enabled, all }: enabled ++ (with all; [ ds ]));
```
and the `infos` returned from `nix.Search` is ` map[packages.x86_64-darwin.php:php-with-extensions-8.1.19]`

However, inside the package's flake.nix, I can make this edit:
```
 php = pkgs.cowsay;
```
and now the `infos` is `map[packages.x86_64-darwin.php:cowsay-3.7.0]` for the same `query`.